### PR TITLE
refactor: Move Connection in ServerFixture to make TestServer shareable

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
A part of https://github.com/influxdata/influxdb_iox/issues/4368 (not starting new processes for each test). 

It turns out that you can't try and make requests over the same `Connection` object, so when I tried to share `TestServer`s which have cached `Connection`s this turned out to be a problem.

# Changes
1. Cache connections on `ServerFixture` rather than `TestServer`

This is the same design as in the OG end to end tests [code link](https://github.com/influxdata/influxdb_iox/blob/main/influxdb_iox/tests/common/server_fixture.rs#L84-L87) which I did not appreciate when I ported this over

* Shared server clusters: alamb/share_server_clusters https://github.com/influxdata/influxdb_iox/issues/4368

### CODE REVIEW

File: test_helpers_end_to_end_ng/src/server_fixture.rs [hunk 1]
```diff
@@ -18,10 +18,12 @@ use crate::{database::initialize_db, server_type::AddAddrEnv};
 use super::{addrs::BindAddresses, ServerType, TestConfig};
 
 /// Represents a server that has been started and is available for
-/// testing.
+/// testing. This structure can not be shared between tests as the
+/// connections are not shareable
```
Comment (carols10cents) [inline test_helpers_end_to_end_ng/src/server_fixture.rs:22]: This sentence sounds a bit like a tautology but I'm not sure how to fix it... not a huge deal

File: test_helpers_end_to_end_ng/src/server_fixture.rs [hunk 2]
```diff
@@ -293,31 +328,32 @@ impl TestServer {
...
-                    return;
-                }
-                ServerState::Ready => {
-                    return;
-                }
-                ServerState::Error => {
-                    panic!("Server was previously found to be in Error, aborting");
                 }
+                ServerState::Ready => {}
+                ServerState::Error => {}
```
Comment (carols10cents) [inline test_helpers_end_to_end_ng/src/server_fixture.rs:347]: Is this correct? The old code would panic if the server was in an error state, and now it continues? Is it possible for the server to get into an error state? Is this not completed yet?

**Review by carols10cents — APPROVED** [review]
Couple of thoughts/questions but nothing blocking.

File: test_helpers_end_to_end_ng/src/server_fixture.rs (hunk 1 above)
Comment (alamb) [reply test_helpers_end_to_end_ng/src/server_fixture.rs:22]: I'll try and make it clearer.

File: test_helpers_end_to_end_ng/src/server_fixture.rs (hunk 2 above)
Comment (alamb) [reply test_helpers_end_to_end_ng/src/server_fixture.rs]: that is an excellent spot -- restored the panic in6fdab5acf

**Pull request merged**